src/Catalog/Registration/RegistrationCatalogEntry.cs

```diff
@@ -1,12 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using VDS.RDF;
 
 namespace NuGet.Services.Metadata.Catalog.Registration
 {
+    public delegate bool ShouldIncludeRegistrationPackage(RegistrationEntryKey key, string resourceUri, IGraph graph);
```

Comment:
I meant to move it inside RegistrationCatalogEntry, but I agree with your logic. However, the name is still confusing and needs a comment/better name.